course6/toyapp-adam/math/transform.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "matrix.h"
+
+namespace Math {
+
+class Transform
+{
+public:
+	void SetPosition(Vector3D position) { mPosition = position; }
+	void SetScale(float size) { mScale = size; }
+	void SetOrientation(Vector3D yawPitchRoll) { mOrientation = yawPitchRoll; }
+
+	void Move(Vector3D delta) { mPosition += delta; }
+	void Grow(float delta) { mScale += delta; }
+	void Rotate(Vector3D yawPitchRoll) { mOrientation += yawPitchRoll; }
+
+	Matrix44 CreateMatrix() const;
+	Matrix44 CreateInverseMatrix() const;
+private:
+	Vector3D mPosition;
+	Vector3D mOrientation;
+	float mScale = 1.0f;
+};
```

Comment:
Do you really need this class? This could simply be the Matrix

Reply:
I think I do, it encapsulates movement, orientation and scale. Which do not belong IN a matrix, but are used to construct matrices. What if I want to e.g. scale an object which is already rotated and positioned somewhere?